finagle-core/src/main/scala/com/twitter/finagle/client/StackClient.scala

```diff
@@ -340,9 +340,14 @@ trait StdStackClient[Req, Rep, This <: StdStackClient[Req, Rep, This]]
       val parameters = Seq(implicitly[Stack.Param[Transporter.EndpointAddr]])
       def make(prms: Stack.Params, next: Stack[ServiceFactory[Req, Rep]]) = {
         val Transporter.EndpointAddr(addr) = prms[Transporter.EndpointAddr]
-        val endpointClient = copy1(params=prms)
-        val transporter = endpointClient.newTransporter()
-        Stack.Leaf(this, ServiceFactory(() => transporter(addr).map(endpointClient.newDispatcher)))
+        val factory = addr match {
+          case ServiceFactorySocketAddress(sf: ServiceFactory[Req, Rep]) => sf
```

Comment:
since these generics are erased, this is implicitly a cast.  could we instead make it:

```scala
  case ServiceFactorySocketAddress(sf: ServiceFactory[_, _]) => sf.asInstanceOf[ServiceFactory[Req, Rep]]
```

so that the cast becomes explicit?